src/go/configinfo/service_info.go

```diff
@@ -504,6 +510,22 @@ func (s *ServiceInfo) processUsageRule() error {
 	return nil
 }
 
+func (s *ServiceInfo) processJwtInQueryParams() {
+	authn := s.serviceConfig.GetAuthentication()
+	for _, provider := range authn.GetProviders() {
```

Comment:
need to do that per Provider.   Default is per provider:  if jwt_locations is defined, use it, otherwise use default.